scripts/freeze.mjs

```diff
@@ -0,0 +1,111 @@
+/**
+ * Code-freeze helper script.
+ * Required to be run on a release branch.
+ *
+ * It takes extends the committed changes with a new version and release date, runs builds and tests and pushes the
+ * changes to the release branch.
+ */
+import { createRequire } from 'module';
+import inquirer from 'inquirer';
+import {
+  cleanNodeModules,
+  displayErrorMessage, scheduleRelease,
+  spawnProcess, verifyBundles
+// eslint-disable-next-line import/extensions
+} from './utils/index.mjs';
+
+// Using `require` exclusively to import `json` files. Prevents 'experimental' warnings from being thrown in the
+// console.
+const require = createRequire(import.meta.url);
+const [version, releaseDate] = process.argv.slice(2);
+
+// Initial verification prompt
+inquirer.prompt([
+  {
+    type: 'confirm',
+    name: 'continueFreeze',
+    message: 'This script will create and commit a code-freeze of the entire repo. \nContinue?',
+    default: true,
+  }
+]).then(async(answers) => {
+  if (!answers.continueFreeze) {
+    return;
+  }
+
+  // Check if we're using the minimum required npm version.
+  await spawnProcess('npm --version', true, (childProcess) => {
+    const npmVersion = childProcess.stdout.toString().replace('\n', '').split('.');
+
+    if (
+      (npmVersion[0] < 7) ||
+      (npmVersion[0] >= 7 && npmVersion[1] < 2)
+    ) {
+      displayErrorMessage('The minimum required npm version is 7.2.0');
+      process.exit(1);
+    }
+  });
+
+  // Check if we're on a release branch.
+  let branchName = null;
+
+  await spawnProcess('git rev-parse --abbrev-ref HEAD', true, (childProcess) => {
+    branchName = childProcess.stdout.toString().replace('\n', '');
+
+    if (!branchName.startsWith('release/')) {
+      displayErrorMessage(
+        'You are not on a release branch. Create a release branch using the `git flow release start`' +
+        ' command on the current `develop` branch.');
+      process.exit(1);
+    }
+  });
+
+  // Check if `git flow` is installed.
+  await spawnProcess('git flow version', true, null, () => {
+    displayErrorMessage('Git flow not installed.');
+  });
+
+  // Check if all the files are committed.
+  await spawnProcess('git status -s', true, (output) => {
+    // If there are any uncommitted changes, kill the script.
+    if (output.stdout.length > 0) {
+      displayErrorMessage('There are uncommitted changes present. Exiting.');
+
+      process.exit(1);
+    }
+  });
+
+  // Bump the version in all packages.
+  if (version && releaseDate) {
+    await scheduleRelease(version, releaseDate);
+
+  } else {
+    await scheduleRelease();
+  }
```

Comment:
```suggestion
  await scheduleRelease(version, releaseDate);
```
The condition is already checked [within the function](https://github.com/handsontable/handsontable/blob/feature/issue-7380-phase-2/scripts/utils/pre-release.mjs#L226). Can we simplify it that way?